From 37e59fb54e88c29dfa1790784e8e62ea315e8792 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 5 Feb 2024 16:01:16 +0100
Subject: [PATCH 551/841] feat(stm32mp25-fdts): add ramp delay on ddr supply

Replace the hardcoded timeout by regulator-enable-ramp-delay for
the DDR main supply (vpp_ddr for DDR4 or vdd1_ddr for LP-DDR4).

This delay is defined to 1ms, the needed time for STPMIC25 LDO
ramp-up. It replace the previous value 2ms for STM32MP25F-DK.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 26067bd248bd04fc2d74b7d5006c008bbbb95050 feat(stm32mp25-fdts): add ramp delay on ddr supply]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I10d387d1c1112a08d98bf9c4a47e5985b9816843
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/357729
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/368535
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 fdts/stm32mp257f-dk.dts  | 1 +
 fdts/stm32mp257f-ev1.dts | 1 +
 2 files changed, 2 insertions(+)

diff --git a/fdts/stm32mp257f-dk.dts b/fdts/stm32mp257f-dk.dts
index 51caf708b..853674ebe 100644
--- a/fdts/stm32mp257f-dk.dts
+++ b/fdts/stm32mp257f-dk.dts
@@ -132,6 +132,7 @@
 				regulator-name = "vdd1_ddr";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
+				regulator-enable-ramp-delay = <1000>;
 			};
 			vdd3v3_usb: ldo4 {
 				regulator-name = "vdd3v3_usb";
diff --git a/fdts/stm32mp257f-ev1.dts b/fdts/stm32mp257f-ev1.dts
index 848ab8992..b91c353bb 100644
--- a/fdts/stm32mp257f-ev1.dts
+++ b/fdts/stm32mp257f-ev1.dts
@@ -143,6 +143,7 @@
 				regulator-name = "vpp_ddr";
 				regulator-min-microvolt = <2500000>;
 				regulator-max-microvolt = <2500000>;
+				regulator-enable-ramp-delay = <1000>;
 			};
 			vdd_sdcard: ldo7 {
 				regulator-name = "vdd_sdcard";
-- 
2.47.2

